application/controllers/Oqrs.php

```diff
@@ -15,17 +15,23 @@ function __construct() {
 		if (($this->config->item('disable_oqrs') ?? false)) { $this->session->set_flashdata('error', __("You're not allowed to do that!")); redirect('dashboard'); }
 	}
 
-    public function index() {
+    public function index($public_slug = NULL) {
 		$this->load->model('oqrs_model');
+		$this->load->model('publicsearch');
 
+      $slug = $this->security->xss_clean($public_slug);
+		$data['slug'] = $slug;
+		$data['oqrs_enabled'] = $this->oqrs_model->oqrs_enabled($slug);
+		$data['public_search_enabled'] = $this->publicsearch->public_search_enabled($slug);
+		$data['disable_oqrs'] = $this->config->item('disable_oqrs');
 		$data['stations'] = $this->oqrs_model->get_oqrs_stations();
 		$data['page_title'] = __("Log Search & OQRS");
 		$data['global_oqrs_text'] = $this->optionslib->get_option('global_oqrs_text');
 		$data['groupedSearch'] = $this->optionslib->get_option('groupedSearch');
 
 		$this->load->view('visitor/layout/header', $data);
 		$this->load->view('oqrs/index');
-		$this->load->view('interface_assets/footer');
+		$this->load->view('interface_assets/footer', $data);
```

Comment:
we should not pass `$data` twice to a page


```suggestion
		$this->load->view('interface_assets/footer');
```